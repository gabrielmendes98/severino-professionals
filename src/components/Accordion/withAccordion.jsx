/* eslint-disable react/display-name */
import PropTypes from 'prop-types';
import Accordion from '@material-ui/core/Accordion';
import AccordionSummary from '@material-ui/core/AccordionSummary';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import AccordionDetails from '@material-ui/core/AccordionDetails';
import Text from 'components/Text';

const withAccordion = Component => {
  const wrappedAccordion = ({
    name,
    handleChange,
    expanded,
    title,
    ...props
  }) => (
    <Accordion
      expanded={expanded === name}
      onChange={handleChange(name)}
      TransitionProps={{ unmountOnExit: true }}
    >
      <AccordionSummary
        expandIcon={<ExpandMoreIcon />}
        aria-controls={`${name}-content`}
        id={`${name}-header`}
      >
        <Text color="secondary">{title}</Text>
      </AccordionSummary>
      <AccordionDetails>
        <Component {...props} />
      </AccordionDetails>
    </Accordion>
  );

  wrappedAccordion.propTypes = {
    name: PropTypes.string.isRequired,
    handleChange: PropTypes.func.isRequired,
    expanded: PropTypes.oneOfType([PropTypes.string, PropTypes.bool])
      .isRequired,
    title: PropTypes.string.isRequired,
  };

  return wrappedAccordion;
};

export default withAccordion;
