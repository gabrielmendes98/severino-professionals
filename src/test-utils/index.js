/* eslint-disable react/prop-types */
import { render, configure, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Form, Formik } from 'formik';
import MainProvider from 'commons/providers/MainProvider';

configure({ testIdAttribute: 'id' });

const Wrapper = ({ children }) => (
  <>
    <MainProvider>{children}</MainProvider>
    <div id="modal"></div>
  </>
);
const RouteWrapper = ({ children }) => (
  <BrowserRouter>
    <Wrapper>{children}</Wrapper>
  </BrowserRouter>
);
const FormikWrapper = ({ children, initialValues, ...config }) => (
  <Formik initialValues={initialValues} {...config}>
    <Form>{children}</Form>
  </Formik>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

const renderWithRouter = (ui, { route = '/', ...options } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: RouteWrapper, ...options });
};

const asFragment = ui => {
  const { asFragment } = customRender(ui);
  return asFragment(ui);
};

const toMatchSnapshot = (Component, { useRouter = false } = {}) => {
  const wrapper = asFragment(
    useRouter ? <BrowserRouter>{Component}</BrowserRouter> : Component,
  );
  expect(wrapper).toMatchSnapshot();
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export {
  customRender as render,
  screen,
  toMatchSnapshot,
  asFragment,
  renderWithRouter,
  FormikWrapper,
};
