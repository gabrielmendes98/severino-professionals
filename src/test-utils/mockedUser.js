const mockedUser = {
  id: '9c6cb22e-f284-4ca4-9a88-d64ece7b896c',
  createdAt: '2021-09-02T03:55:52.338Z',
  updatedAt: '2021-09-02T03:55:52.338Z',
  email: 'test@test.com',
  name: 'Testerson da Silva',
  description: null,
  avatarUrl: 'www.avatarurlfake.com',
  phone: '(11) 11111-1112',
  state: 'Minas Gerais',
  city: '√Åguas Formosas',
  hasWhatsapp: false,
};
export default mockedUser;
