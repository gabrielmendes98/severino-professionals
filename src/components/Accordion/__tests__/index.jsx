import { toMatchSnapshot } from 'test-utils';
import withAccordion from '../withAccordion';

it('should match snapshot when accordion is closed', () => {
  const TestComponent = () => <div>Test</div>;
  const Component = withAccordion(TestComponent);
  toMatchSnapshot(
    <Component
      name="test"
      title="Test Accordion"
      expanded={'test2Expanded'}
      handleChange={() => {}}
    />,
  );
});

it('should match snapshot when accordion is opened', () => {
  const TestComponent = () => <div>Test</div>;
  const Component = withAccordion(TestComponent);
  toMatchSnapshot(
    <Component
      name="test"
      title="Test Accordion"
      expanded={'test'}
      handleChange={() => {}}
    />,
  );
});
