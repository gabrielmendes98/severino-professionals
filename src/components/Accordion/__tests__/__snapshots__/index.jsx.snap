// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot when accordion is closed 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  color: #3700B3;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: Roboto;
}

<div
    class="MuiPaper-root MuiAccordion-root MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="test-content"
      aria-disabled="false"
      aria-expanded="false"
      class="MuiButtonBase-root MuiAccordionSummary-root"
      id="test-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content"
      >
        <p
          class="MuiTypography-root c0 MuiTypography-body1"
          textcolor="secondary"
          transform=""
        >
          Test Accordion
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </div>
  <div
    id="modal"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`should match snapshot when accordion is opened 1`] = `
<DocumentFragment>
  .c0 {
  margin-top: 0px;
  margin-left: 0px;
  margin-right: 0px;
  margin-bottom: 0px;
  color: #3700B3;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-family: Roboto;
}

<div
    class="MuiPaper-root MuiAccordion-root Mui-expanded MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      aria-controls="test-content"
      aria-disabled="false"
      aria-expanded="true"
      class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded"
      id="test-header"
      role="button"
      tabindex="0"
    >
      <div
        class="MuiAccordionSummary-content Mui-expanded"
      >
        <p
          class="MuiTypography-root c0 MuiTypography-body1"
          textcolor="secondary"
          transform=""
        >
          Test Accordion
        </p>
      </div>
      <div
        aria-disabled="false"
        aria-hidden="true"
        class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon Mui-expanded MuiIconButton-edgeEnd"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
    <div
      class="MuiCollapse-root MuiCollapse-entered"
      style="min-height: 0px;"
    >
      <div
        class="MuiCollapse-wrapper"
      >
        <div
          class="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby="test-header"
            id="test-content"
            role="region"
          >
            <div
              class="MuiAccordionDetails-root"
            >
              <div>
                Test
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    id="modal"
  />
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
